openfisca_aotearoa/variables/acts/social_security/supported_living_payment/supported_living_payment.py

```diff
@@ -1,124 +1,236 @@
+        return ssa40B + ssa40D
 
-class supported_living_payment__receiving(variables.Variable):
+    def formula_2018_11_26(persons, period, parameters):
+        """
+        For the 2018 version of the act, this is split into two sections:
+            34 - 39 on ground of restricted work capacity or total blindness
+            40 - 42 on ground of caring for another person
+        """
+        ssa34 = persons("supported_living_payment__disabled_or_blind__entitled", period, parameters)
+        ssa40 = persons("supported_living_payment__carer__entitled", period, parameters)
+
+        # ssa116 - must undergo work ability assessment
+
+        # ssa117 - unless they are terminally ill or their condition will not "improve"
+
+        # ssa121_e - spouse must undertake work-preparation obligations
+
+        # ssa128 - children must atend school & healthcare, person required to attend interviews
+
+        # ssa140 - spouse must be available to work
+
+        # ssa170 - person or spouse must work with service providers
+
+        return ssa34 + ssa40
+
+
+class supported_living_payment__disabled_or_blind__entitled(variables.Variable):
     value_type = bool
-    default_value = False
     entity = entities.Person
-    label = "Person is currently recieving/being paid the supported living payment"
     definition_period = periods.WEEK
-    reference = "Reference is unclear, but concept underpinning the variable assumes it covers both: 'being paid a main benefit' or 'recieving a benefit'"
+    label = "Eligible for Supported Living Payment, on ground of restricted work capacity or total blindness."
+    reference = [
+        "https://www.legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783174",
+        ]
 
+    def formula_2018_11_26(persons, period, parameters):
+        # person P has restricted work capacity or is totally blind
+        ssa34_a = persons("supported_living_payment__restricted_work_capacity", period.first_week) + persons("totally_blind", period.start)
 
-class supported_living_payment__caring_for_another_person(variables.Variable):
+        # person P meets the residential requirement
+        ssa34_b = persons("social_security__residential_requirement", period.first_week)
+
+        # person P is 16 or older
+        ssa34_c = persons("age", period.start) >= 16
+
+        # ssa35 specifies requirements for restricted work capacity status
+
+        # person P is ineligible if the disability was self-inflicted to gain benefits
+        ssa36 = numpy.logical_not(persons("supported_living_payment__disability_self_inflicted", period))
+
+        # ssa37 specifies medical examination for reassessment of work capacity
+
+        # ssa38 specifies payment rates for person P and partner / spouse S if applicable
+
+        # ssa39 specifies open employment is permitted for reassessment of work capacity
+
+        return ssa34_a * ssa34_b * ssa34_c * ssa36
+
+    def formula(persons, period, parameters):
```

Comment:
This 40B section appears to be referring to the 1964 Act. The formula definition possibly needs a date as it's not clear when this formula is required or that it's an older version. If the 1964 act logic is required (I'm not yet clear why it is) we should also had a link to it in the reference above based on the appropriate dated version

Reply:
Just to clarify, I really appreciate the 1964 Act being addressed but not sure it's a requirement right now.